packages/data/lib/repository/movie_repository.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:data/service/api_base_service.dart';
+import 'package:data/service/service_payload.dart';
+import 'package:domain/di/data_model.dart';
+import 'package:domain/repository/base_repository.dart';
+
+class NetworkRepositoryImpl implements NetworkRepository {
+  final ApiBaseService<ServicePayload> _apiService;
+
+  NetworkRepositoryImpl(this._apiService);
+
+  @override
+  Future<GetDataResponse> getData({
+    required String apiPath,
+    int? itemCount,
+  }) async {
+    return _apiService.get(
+      apiPath,
+      queryParameters: {
+        'extended': 'full',
```

Comment:
move the query parameters to the constants file
you have the first and second request sent by "extended: full", although only the first should